### ISSUE TITLE
Rule Request: Enforce `Double` for function arguments instead of `CGFloat`

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#2753]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

> 1. Why should this rule be added? Share links to existing discussion about what the community thinks about this.

I prefer to only use `Double` for floats in my own functions, for consistency and ease of use, never `CGFloat`. I only support 64-bit systems, so `CGFloat` has no practical benefit for me. However, I often get PRs on my open source projects where they use `CGFloat`. Would be nice to be able to enforce using `Double`.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Would trigger:

```swift
func foo(number: CGFloat) {}
```

Would not trigger:

```swift
func foo(number: Double) {}
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

No

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in

**Comment by cdstamper** [comment]
@sindresorhus is there a reason to target function arguments specifically, and not a CGFloat used elsewhere?

**Comment by sindresorhus** [comment]
It's the easiest case to target that will never have false-positives. Let's say the rule also supports variable declarations. What if a variable is passed into an Apple API that requires CGFloat? You would have to do `appleAPI(number: CGFloat(yourNumber))` at every callsite. While I personally don't mind that for the purity of having Double's everywhere, that's more controversial, and I wanted to get the least controversial thing added first.
